k8s-infra-configure.yml

```diff
@@ -0,0 +1,21 @@
+---
+#####
+## Performs Kubernetes cluster configuration for any inventory that defines
+## the groups "kube-masters" and "kube-workers"
+#####
+
+# Do cluster setup for all hosts
+- hosts: kube-masters,kube-workers
+  #remote_user: root
+  roles:
+    - jasmin.k8s/kubeadm
```

Comment:
We could run this role using:

```
roles:
  - role: jasmin.k8s/kubeadm
    vars:
      kube_role: "{% if inventory_hostname in groups['kube-masters'] %}master{% else %}worker{% endif %}"
```

This would avoid having the single-line `group_vars` files. Not sure which I prefer.

Reply:
The alternative would be to modify the `kubeadm` role so that instead of `kube_role: {master|worker}` it would have two booleans, `kube_role.master = true|false` and `kube_role.worker = true|false`.

Then this would become:

```
roles:
  - role: jasmin.k8s/kubeadm
    kube_role:
      master: "{{ inventory_hostname in groups['kube-masters'] }}"
      worker: "{{ inventory_hostname in groups['kube-workers'] }}"
```

It would also mean all the `when: "kube_role == "master|worker"` statements could be replaced by `when: kube_role.master` or `when: kube_role.worker`.

Both seem nicer. I think we should do this.

Reply:
What about: `kube_role: {% 'master' if inventory_hostname in groups['kube-masters'] else 'worker' %}`

Reply:
Because there is a section in k8s-ansible playbook that uses the value of kube_role.

Reply:
To be precise `roles/jasmin.k8s/kubeadm/tasks/prerequisites.yml:67:    - "kubernetes-{{ kube_role }}"`

Reply:
That is true. But your suggestion has less mess. Although I think you need `{{` and `}}` rather than `{%` and `%}`.

Reply:
Ah yes, I've addressed all your comments now, I think.